docs/src/main/asciidoc/security-built-in-authentication.adoc

```diff
@@ -0,0 +1,163 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - BuiltIn Authentication Support
+
+include::./attributes.adoc[]
+
+This document describes the Quarkus built in authentication mechanisms for HTTP based FORM, BASIC and Mutual TLS authentication as well as the proactive authentication.
+
+[[basic-auth]]
+== Basic Authentication
+
+To enable basic authentication set `quarkus.http.auth.basic=true`. You must also have at least one extension installed
+that provides a username/password based `IdentityProvider`, such as link:security-jdbc[Elytron JDBC].
```

Comment:
Maybe list the core extensions that provide that with links to their respective docs?

Reply:
Hi @gastaldi, the current [Secuity page](https://quarkus.io/guides/security) has a large space gap due to a table (which mixes different not very related guides together as well) so I thought, so I dropped the `table` concept and decided to go with small subsections giving a bit more context and then linking to the guides. 
I copied this text from the current `security.adoc`, in this PR `security.adoc` has an `Identity Providers` section (I can add the table here but since we only 2 standalone identity providers, I'm not sure, but can be done easily when we finalize it), I think I can link back to that section from here, let me do it.